src/coreclr/src/gc/gc.cpp

```diff
@@ -22032,7 +22311,40 @@ void gc_heap::plan_phase (int condemned_gen_number)
         )
     {
 #ifndef MULTIPLE_HEAPS
-        _sort (&mark_list[0], mark_list_index-1, 0);
+#ifdef USE_VXSORT
+        ptrdiff_t entry_count = mark_list_index - mark_list;
+        // conservatively use AVX2 only for large mark lists,
+        // and do runtime test to check whether AVX2 is indeed available 
+        if (entry_count > AVX2_THRESHOLD_SIZE && IsSupportedInstructionSet (InstructionSet::AVX2))
+        {
+            int32_t* mark_list_32 = (int32_t*)mark_list;
+            uint8_t* low = slow;
+            ptrdiff_t range = shigh - low;
+            if ((uint32_t)range == range)
+            {
+                do_pack_avx2 (mark_list, entry_count, low);
+                _sort (&mark_list_32[0], &mark_list_32[entry_count - 1], 0);
+                do_unpack_avx2 (mark_list_32, entry_count, low);
```

Comment:
can we get the pack/int32sort/unpack in one method, something like try_vxsort_32 so we can call it both in `plan_phase` and `sort_mark_list`?

Reply:
That's a good idea - in fact I moved the code that checks the range, the instruction set etc. into one method do_vxsort.